airbyte-integrations/connectors/source-netsuite/source_netsuite/streams.py

```diff
@@ -0,0 +1,266 @@
+            # and null has not already been added
+            if property_type and not isinstance(property_type, dict) and "null" not in property_type_list:
+                record["type"] = ["null"] + property_type_list
+            # removing non-functional elements from schema
+            for element in remove_from_schema:
+                if record.get(element):
+                    record.pop(element)
+
+            ref = record.get("$ref")
+            if ref:
+                ns_link = ref == "/services/rest/record/v1/metadata-catalog/nsLink"
+                return self.get_schema(ref) if ns_link else self.ref_schema()
+            else:
+                return {k: self.build_schema(v) for k, v in record.items()}
+        else:
+            return record
+
+    def get_json_schema(self, **kwargs) -> dict:
+        schema = self.get_schema(META_PATH + self.name)
+        return self.build_schema(schema)
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        resp = response.json()
+        has_more = resp.get("hasMore")
+        if has_more:
+            return {"offset": resp["offset"] + resp["count"]}
+        return None
+
+    def format_date(self, last_modified_date: str) -> str:
+        # the date format returned is differnet than what we need to use in the query
+        lmd_datetime = datetime.strptime(last_modified_date, self.output_datetime_format)
+        return lmd_datetime.strftime(self.input_datetime_format)
+
+    def request_params(self, next_page_token: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        params = {}
+        if next_page_token:
+            params.update(**next_page_token)
+        return params
+
+    def fetch_record(self, record: Mapping[str, Any], request_kwargs: Mapping[str, Any]) -> Iterable[Mapping[str, Any]]:
+        url = record["links"][0]["href"]
+        args = {"method": "GET", "url": url, "params": {"expandSubResources": True}}
+        prep_req = self._session.prepare_request(requests.Request(**args))
+        response = self._send_request(prep_req, request_kwargs)
+        # sometimes response.status_code == 400,
+        # but contains json elements with error description,
+        # to avoid passing it as TYPE: RECORD, we filter response by status
+        if response.status_code == requests.codes.ok:
+            yield response.json()
+
+    def parse_response(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+        **kwargs,
+    ) -> Iterable[Mapping]:
+
+        records = response.json().get("items")
+        request_kwargs = self.request_kwargs(stream_slice, next_page_token)
+        if records:
+            for record in records:
+                # make sub-requests for each record fetched
+                yield from self.fetch_record(record, request_kwargs)
```

Comment:
shouldn't this be a substream? to handle retries etc

Reply:
Great idea, will handle this in BETA